### ISSUE TITLE
the difficulty problems of samtools in my macbook pro

### ISSUE THREAD

**Issue by Ethanlee20192019** [samtools/htslib#1312]
i don't know how to solve this problem. i have seen the same problems in github, and i also try my best and many ways to solve it, but failed. i need help and new code to solve it. thanks everyone
(dna) juliama@Ethan-leedeMacBook-Pro ~ % samtools
dyld: Library not loaded: @rpath/libdeflate.so
  Referenced from: /Users/Ethan/opt/anaconda3/envs/dna/bin/samtools
  Reason: image not found
zsh: abort      samtools

**Comment by daviesrob** [comment]
It looks like you may have found an instance of [this bioconda-recipes issue](https://github.com/bioconda/bioconda-recipes/issues/15415).  It's not entirely clear of this has been fixed - which version of `samtools` were you trying to build?

One possible work-around (suggested in the linked issue) you could try would be to find where anaconda has put `libdeflate.dylib`, and then `cd` to that directory and make a symbolic link:
```
ln -s libdeflate.dylib libdeflate.so
```

With a bit of luck `samtools` would be able to use it to find the right file.

**Comment by jmarshall** [comment]
That bioconda issue was fixed in October 2019 in htslib-1.9-h356306b_9 and samtools-1.9-h8aa4d43_12 by bioconda/bioconda-recipes#17273 and bioconda/bioconda-recipes#18322. However the nature of Conda is that older packages remain available for installation should you choose to do so or if you are unlucky enough that they are the solution to the package equation implied by what is already installed in the environment in which you are installing them.

So the first question is what version of bioconda's samtools package you have, which can be answered with `conda list|egrep '(samtools|htslib)'`. If you are having this error, then most likely what your environment contains is an extremely outdated version of samtools.

But the real question is what version of samtools you **want**. If you actually want to use one of these old versions, that is one thing; but almost certainly what you would prefer is an up-to-date version, in which case any advice around symlinking _libdeflate_ libraries (as suggested on that issue) is extremely outdated.

Instead, you should install samtools into a fresh conda environment with

```
conda create -n dna-samtools samtools
```

which will install the current version (at the time of writing, 1.13) and correct forms of its dependencies. Then you can either use that environment (probably the best approach), or use it as a guide for what to aim for when updating packages in your `dna` environment to match.

**Comment by jkbonfield** [comment]
I really wouldn't recommend trying to work around conda issues by symlink hackery.  It's just going to make a mess even messier!

Fundamentally though, this isn't an htslib issue so I'm closing it.  If you have problems with conda packages, please raise them on the bioconda github project.  It's important for them to track such issues and they won't be monitoring all the individual package githubs.
